docs/adr/adr-008-monitoring-and-alerting.md

```diff
@@ -0,0 +1,269 @@
+# 8. Monitoring and alerting
+
+Date: 2020-11-13
+
+## Context
+
+Email Alert API has acquired the undesirable reputation of an application that
+produces a high volume of unactionable alerts and is a frequent source of
+GOV.UK incidents. The monitoring and alerting approach no longer appears to
+be sufficient for the application behaviour as the application's volume of
+work has changed in the past 2 years. In previous years, the application
+would reach a peak of sending 1 million emails a day whereas now it can be
+sending over 10 million emails a day. This elevated volume of work has
+increased system latency which has impacted alerts and contributed to
+incidents.
+
+The approach used to monitoring the application and the wider email alert
+system is two fold. One approach is to monitor Email Alert API's internal
+state to see if it is [completing tasks within a particular
+time frame][monitor-time-frame] or [whether there is a high quantity of work to
+do][monitor-workload]. The other approach is to use an [end-to-end
+check][email-alert-monitoring] which verifies that a Gmail account
+has received an email within a time frame. Both of these approaches are flawed.
+
+Monitoring for high quantities of work, and whether work was completed in a
+time frame, has left Email Alert API vulnerable to producing alerts when the
+application is busy, yet fully operational. These alerts are unactionable for
+support engineers as the alerts resolve themselves once the work is completed
+and there is little an engineer can do to speed this process up. For example,
+if the application has accumulated 1,000,000 emails to send it raises a
+critical alert even if it is sending these at the [fastest rate Notify
+would allow][rate-limit]. These alerts, therefore, waste the time of support
+engineers and create a risk that a support engineer might miss an alert that
+requires their intervention.
+
+Since these alerts can't be relied upon to reliably suggest the
+application is broken, they aren't used for any out-of-hours alerts. Therefore,
+even if they did reflect a genuine problem no engineers would be contacted.
+An engineer instead learns of a problem out-of-hours due to the other type
+of monitoring approach: the end-to-end checks. These alert when content
+published through [Travel Advice Publisher][] does not arrive in
+a particular Gmail inbox within a specified time period.
+
+Monitoring an inbox for a limited set of emails has its own distinct flaws.
+If Email Alert API breaks outside of office hours then engineers would not be
+notified unless travel advice content was published. This alert also monitors
+beyond the boundaries of the email alert system - considering whether
+GOV.UK publishing, Notify and Gmail are all working - leading to alerts
+representing problems outside the control of GOV.UK engineers (such as Gmail
```

Comment:
```suggestion
representing problems that are irrelevant to the system, or outside the control of GOV.UK engineers (such as Gmail
```

Minor: previously I didn't clock that we're implying problems with GOV.UK Publishing might be outside the control of GOV.UK engineers.